ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,35 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(ARG "" "" "REQUIRED_PACKAGES" ${ARGN})
+  if(ARG_UNPARSED_ARGUMENTS)
     message(FATAL_ERROR "ament_auto_find_build_dependencies() called with "
-      "unused arguments: ${ARGN}")
+      "unused arguments: ${ARG_UNPARSED_ARGUMENTS}")
   endif()
 
   if(NOT _AMENT_PACKAGE_NAME)
     ament_package_xml()
   endif()
 
+  # ensure that the caller isn't expecting additional packages to be found
+  set(_build_and_buildtool_depends
+    ${${PROJECT_NAME}_BUILD_DEPENDS}
+    ${${PROJECT_NAME}_BUILDTOOL_DEPENDS})
+  foreach(_package_name ${ARG_REQUIRED_PACKAGES})
+    if(NOT _package_name IN_LIST _build_and_buildtool_depends)
+      message(FATAL_ERROR "ament_auto_find_build_dependencies() called with "
+      "required package that is not listed as a build/buildtool dependency in "
+      "the package.xml: ${_package_name}")
```

Comment:
Please indent the two lines with two spaces since they are part of the function call.